src/top.rs

```diff
@@ -0,0 +1,124 @@
+use crate::config;
+use prettytable::{format, Table};
+use structopt::StructOpt;
+use url::Url;
+use vector_api_client::{
+    gql::{HealthQueryExt, TopologyQueryExt},
+    Client,
+};
+
+trait StatsWriter {
+    fn kb(&mut self, n: f64) -> String;
+}
+
+struct HumanWriter {
+    f: human_format::Formatter,
+}
+
+impl HumanWriter {
+    fn new() -> Self {
+        Self {
+            f: human_format::Formatter::new(),
+        }
+    }
+}
+
+impl StatsWriter for HumanWriter {
+    fn kb(&mut self, n: f64) -> String {
+        self.f.with_decimals(2).format(n)
+    }
+}
+
+struct LocaleWriter {
+    buf: num_format::Buffer,
+}
+
+impl LocaleWriter {
+    fn new() -> Self {
+        Self {
+            buf: num_format::Buffer::new(),
+        }
+    }
+}
+
+impl StatsWriter for LocaleWriter {
+    fn kb(&mut self, n: f64) -> String {
+        self.buf
+            .write_formatted(&(n as i64), &num_format::Locale::en);
+        self.buf.to_string()
+    }
+}
+
+fn new_formatter(humanize: bool) -> Box<dyn StatsWriter> {
+    if humanize {
+        Box::new(HumanWriter::new())
+    } else {
+        Box::new(LocaleWriter::new())
+    }
+}
+
+#[derive(StructOpt, Debug)]
+#[structopt(rename_all = "kebab-case")]
+pub struct Opts {
+    /// How often the screen refreshes (in milliseconds)
+    #[structopt(default_value = "500", short = "i", long)]
+    refresh_interval: i32,
+
+    #[structopt(short, long)]
+    url: Option<Url>,
+
+    #[structopt(short, long)]
+    humanize: bool,
```

Comment:
I am not sure we need to say "humanize". Seems `ls` uses `--human-readable` but I think that's quite long...

```suggestion
    human: bool,
```

Maybe?